scm/driver/gitlab/release.go

```diff
@@ -0,0 +1,111 @@
+package gitlab
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/drone/go-scm/scm"
+)
+
+type releaseService struct {
+	client *wrapper
+}
+
+type release struct {
+	Title       string `json:"name"`
+	Description string `json:"description"`
+	Tag         string `json:"tag_name"`
+	Commit      struct {
+		ID string `json:"id"`
+	} `json:"commit"`
+}
+
+type releaseInput struct {
+	Title       string `json:"name"`
+	Description string `json:"description"`
+	Tag         string `json:"tag_name"`
+}
+
+func (s *releaseService) Find(ctx context.Context, repo string, id int) (*scm.Release, *scm.Response, error) {
+	// this could be implemented by List and filter but would be to expensive
+	panic("gitlab only allows to find a release by tag")
+}
+
+func (s *releaseService) FindByTag(ctx context.Context, repo string, tag string) (*scm.Release, *scm.Response, error) {
+	path := fmt.Sprintf("api/v4/projects/%s/releases/%s", encode(repo), tag)
+	out := new(release)
+	res, err := s.client.do(ctx, "GET", path, nil, out)
+	return convertRelease(out), res, err
+}
+
+func (s *releaseService) List(ctx context.Context, repo string, opts scm.ReleaseListOptions) ([]*scm.Release, *scm.Response, error) {
+	path := fmt.Sprintf("api/v4/projects/%s/releases", encode(repo))
+	out := []*release{}
+	res, err := s.client.do(ctx, "GET", path, nil, &out)
+	return convertReleaseList(out), res, err
+}
+
+func (s *releaseService) Create(ctx context.Context, repo string, input *scm.ReleaseInput) (*scm.Release, *scm.Response, error) {
+	path := fmt.Sprintf("api/v4/projects/%s/releases", encode(repo))
+	in := &releaseInput{
+		Title:       input.Title,
+		Description: input.Description,
+		Tag:         input.Tag,
+	}
+	out := new(release)
+	res, err := s.client.do(ctx, "POST", path, in, out)
+	return convertRelease(out), res, err
+}
+
+func (s *releaseService) Delete(ctx context.Context, repo string, id int) (*scm.Response, error) {
+	// this could be implemented by List and filter but would be to expensive
+	panic("gitlab only allows to delete a release by tag")
```

Comment:
use `ErrNotImplemented` instead of panic

Reply:
fixed